src/hooks/useOrganization.ts

```diff
@@ -0,0 +1,44 @@
+import { useQuery } from "@tanstack/react-query";
+import { useTranslation } from "react-i18next";
+import { toast } from "sonner";
+
+import query from "@/Utils/request/query";
+import { Organization } from "@/types/organization/organization";
+import organizationApi from "@/types/organization/organizationApi";
+
+interface UseOrganizationParams {
+  orgType?: string;
+  parentId?: string;
+  name?: string;
+  enabled?: boolean;
+}
+
+export function useOrganization({
+  orgType = "",
+  parentId = "",
+  name = "",
+  enabled = true,
+}: UseOrganizationParams) {
+  const { data, isLoading, isError } = useQuery<{ results: Organization[] }>({
+    queryKey: ["organization", orgType, name, parentId],
+    queryFn: query(organizationApi.list, {
+      queryParams: {
+        org_type: orgType,
+        parent: parentId,
+        name,
+      },
+    }),
+    enabled: enabled && !!name,
+    select: (res) => ({ results: res.results }),
+  });
```

Comment:
type would anyways be inferred from the API's route definition right? And what does the `select` even do? it's taking in results and returning results, why do we even need it?

```suggestion
  const { data, isLoading, isError } = useQuery({
    queryKey: ["organization", orgType, name, parentId],
    queryFn: query(organizationApi.list, {
      queryParams: {
        org_type: orgType,
        parent: parentId,
        name,
      },
    }),
    enabled: enabled && !!name,
  });
```

Reply:
Overriding these in #9711 👍